### DISCUSSION TITLE
Slow previous search can replace current search

### DISCUSSION THREAD

**Discussion by Friss** [community/community#10841] [Other Feature Feedback, Questions, & Ideas] [answered]
I had this bug happen once and its been hard to reproduce but figured I would report it. 

I had used the prompt to search a fork in the new codesearch and it was a search for `type` in the repo `HubSpot/opengrok` https://cs.github.com/HubSpot/opengrok?q=Type

As that was loading I changed the search to be for `typescript` and got results right away and then a few seconds later it was replaced with the results for my original search (`type`) but the search input and the url had the search as `typescript`. 

So steps would be 

1) Search for `foo`
2) `foo` search api call takes time 
3) Search for `bar` 
4) Results for `bar` load
5) Results for `foo` load and replace `bar` results.

**Comment by colinwm** [comment, marked as answer]
@Friss thanks for the report! I've fixed this, it shouldn't happen again.
